product_variant_configurator/models/product_configurator.py

```diff
@@ -0,0 +1,272 @@
+            for val in attr_values_dict:
+                val['product_tmpl_id'] = product.product_tmpl_id.id
+                val['owner_model'] = self._name
+                val['owner_id'] = self.id
+            attr_values = [(0, 0, values) for values in attr_values_dict]
+            res['product_attribute_ids'] = attr_values
+            res['product_tmpl_id'] = product.product_tmpl_id.id
+            res['name'] = self._get_product_description(
+                product.product_tmpl_id, product,
+                product.attribute_value_ids)
+        return res
+
+    @api.model
+    def _order_attributes(self, template, product_attribute_values):
+        res = template._get_product_attributes_dict()
+        res2 = []
+        for val in res:
+            value = product_attribute_values.filtered(
+                lambda x: x.attribute_id.id == val['attribute_id'])
+            if value:
+                val['value_id'] = value
+                res2.append(val)
+        return res2
+
+    @api.model
+    def _get_product_description(self, template, product, product_attributes):
+        name = product and product.name or template.name
+        extended = self.user_has_groups(
+            'product_variant_configurator.'
+            'group_product_variant_extended_description')
+        if not product_attributes and product:
+            product_attributes = product.attribute_value_ids
+        values = self._order_attributes(template, product_attributes)
+        if extended:
+            description = "\n".join(
+                "%s: %s" %
+                (x['value_id'].attribute_id.name, x['value_id'].name)
+                for x in values)
+        else:
+            description = ", ".join([x['value_id'].name for x in values])
+        if not description:
+            return name
+        return ("%s\n%s" if extended else "%s (%s)") % (name, description)
+
+    @api.multi
+    def unlink(self):
+        """Mimic `ondelete="cascade"`."""
+        attributes = self.mapped("product_attribute_ids")
+        result = super(ProductConfigurator, self).unlink()
+        if result:
+            attributes.unlink()
+        return result
+
+    @api.model
+    def check_configuration_validity(self, vals):
+        """The method checks that the current selection values are correct.
+
+        As default, the validity means that all the attributes
+        values are set. This can be overridden to set another rules.
+
+        :param vals: Dictionary of values that creates the record
+        :type vals: dict
+        :raises: exceptions.ValidationError: If the check is not valid.
+        """
+        if any(not x[2].get('value_id') for
+               x in vals.get('product_attribute_ids', [])):
+            raise exceptions.ValidationError(
+                _("You have to fill all the attributes values."))
+
+    @api.multi
+    def _create_variant_from_vals(self, vals):
```

Comment:
I have no idea about this : I just ran a `git grep _create_variant_from_vals` in the odoomrp-wip repository, and don't really know these modules :)